diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
index 46487c4..6575381 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java
@@ -42,6 +42,7 @@ import com.facebook.react.views.text.TextAttributes;
 import com.facebook.react.views.text.TextInlineImageSpan;
 import com.facebook.react.views.view.ReactViewBackgroundManager;
 import java.util.ArrayList;
+import java.util.Arrays;
 
 /**
  * A wrapper around the EditText that lets us better control what happens when an EditText gets
@@ -360,10 +361,37 @@ public class ReactEditText extends EditText {
   @Override
   public void setInputType(int type) {
     Typeface tf = super.getTypeface();
-    super.setInputType(type);
-    mStagedInputType = type;
     // Input type password defaults to monospace font, so we need to re-apply the font
     super.setTypeface(tf);
+    int inputType = type;
+
+    // Set InputType to TYPE_CLASS_TEXT (the default one for Android) to fix a crash on Xiaomi
+    // devices with Android Q. This crash happens when focusing on a email EditText within a
+    // ScrollView, a prompt will be triggered but the system fail to locate it properly.
+    // Here is an example post discussing about this issue:
+    // https://github.com/facebook/react-native/issues/27204
+    String[] xiaomiManufacturers = new String[]{
+      "M2002J9G",
+      "M2002J9G",
+      "M2004J19C",
+      "M2007J20CG",
+      "Mi",
+      "MI",
+      "MIX",
+      "POCO",
+      "POCOPHONE",
+      "Redmi",
+      "Xiaomi"
+    };
+    boolean isXiaomi = Arrays.asList(xiaomiManufacturers).contains(Build.MANUFACTURER);
+    if (inputType == InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS
+        && Build.VERSION.SDK_INT == Build.VERSION_CODES.Q
+        && isXiaomi)) {
+      inputType = InputType.TYPE_CLASS_TEXT;
+    }
+
+    super.setInputType(inputType);
+    mStagedInputType = inputType;
 
     /**
      * If set forces multiline on input, because of a restriction on Android source that enables
